great_expectations/agent/actions/run_onboarding_data_assistant.py

```diff
@@ -0,0 +1,86 @@
+from typing import TYPE_CHECKING
+
+from great_expectations.agent.actions.action import (
+    ActionResult,
+    AgentAction,
+    CreatedResource,
+)
+from great_expectations.agent.models import RunOnboardingDataAssistantEvent
+from great_expectations.exceptions import StoreBackendError
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+
+
+class RunOnboardingDataAssistantAction(AgentAction):
+    def run(self, event: RunOnboardingDataAssistantEvent, id: str) -> ActionResult:
+        expectation_suite_name = f"{event.data_asset_name} onboarding assistant suite"
+        checkpoint_name = f"{event.data_asset_name} onboarding assistant checkpoint"
+
+        # ensure resources we create don't already exist
+        try:
+            self._context.get_expectation_suite(
+                expectation_suite_name=expectation_suite_name
+            )
+            raise ValueError(
+                f"Onboarding Assistant Expectation Suite `{expectation_suite_name}` already exists. "
+                + "Please rename or delete suite and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        try:
+            self._context.get_checkpoint(name=checkpoint_name)
+            raise ValueError(
+                f"Onboarding Assistant Checkpoint `{checkpoint_name}` already exists. "
+                + "Please rename or delete Checkpoint and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        datasource = self._context.get_datasource(datasource_name=event.datasource_name)
+        asset = datasource.get_asset(asset_name=event.data_asset_name)
+        batch_request = asset.build_batch_request()
+
+        data_assistant_result = self._context.assistants.onboarding.run(
+            batch_request=batch_request,
+        )
+        expectation_suite: ExpectationSuite = (
+            data_assistant_result.get_expectation_suite(
+                expectation_suite_name=expectation_suite_name
+            )
+        )
+        self._context.add_or_update_expectation_suite(
```

Comment:
Why are we using `add_or_update_*()` for these given we just checked to ensure the `expectation_suite` and `checkpoint` did not exist?

Reply:
sure, this is updated in [bd77fb9](https://github.com/great-expectations/great_expectations/pull/7882/commits/bd77fb9d20972ec8ba1c2dff7eee5d308f6fb4c7). `CloudContext` doesn't have `add_checkpoint` method though [per the docs](https://docs.greatexpectations.io/docs/reference/api/data_context/CloudDataContext_class) so i'll keep using `add_or_update`

Reply:
It's inherited from `AbstractDataContext`
https://docs.greatexpectations.io/docs/reference/api/data_context/abstractdatacontext_class/#great_expectations.data_context.AbstractDataContext